internal/osbuild2/device.go

```diff
@@ -1,8 +1,6 @@
 package osbuild2
 
-type Devices interface {
-	isStageDevices()
-}
```

Comment:
Hmm. I'm probably a bit confused, but if Devices are very similar to Inputs, I would expect `Devices` to be kept as an interface type. Then specific device types would implement it, e.g. the `LoopbackDevice`. And lastly stages accepting the `LoopbackDevice` would have their own stage-specific interface type for devices, which would be then implemented by all Device types which are acceptable for the stage.

As I understand it with Inputs, the Inputs object properties are specific to the Input type. Devices seem to be a bit different and it seems that the `Devices` object can have only `device` property (https://github.com/osbuild/osbuild/blob/569697e9a9fff55c62c694a28120a26c6a7d3c7e/schemas/osbuild2.json#L19-L37). Thus defining `Devices` type as map is IMO too generic.

Reply:
`device` in the schema is just the name for the definition. Devices in osbuild are indeed a `name: object` mapping with an arbitrary number of devices. Some stages (`mkfs*`) require there to be at least one known one (usually called `device`) which is the one they operate on.
I am a bit of an outsider to the go code but on a first glance the map looks fine to me.

Reply:
One other note: I think devices and mounts are simpler than inputs. The latter has `references`, which exact contents is  flexible, i.e. either `array` or `object`, and the contents of said `object` depends on the input type and origin: it is, e.g. different for `org.osbuild.files` input with `pipeline` vs. `org.osbuild.files` with `source` vs. `org.osbuild.tree`.

Reply:
Yes, in the end we agreed that the stage devices and inputs should be represented as map. My comment was caused mostly by a confusion 😃